app/controllers/instruments_dashboard_controller.rb

```diff
@@ -26,6 +26,7 @@ def init_reservations
       .current_in_use
       .merge(Product.alphabetized)
       .includes(:product, order: :user)
+      .sort_by { |r| r.product.schedule.position || 9999 } # there's no direct relation between reservation and schedule, and we want nil's last
```

Comment:
I tried this locally and it seemed to work and might make things simpler and easier to understand.

```ruby
# Reservation
# Allows us to do a `reservations.joins(instrument: :schedule)` later
belongs_to :instrument, foreign_key: :product_id

# Schedule
# Even if we don't go this whole route, I think making this scope will make it read clearer where it's used.
scope :positioned, -> { order_by_asc_nulls_last(:position) }

# Here
@reservations = current_facility.reservations
      .current_in_use
      # Make sure to get rid of `.merge(Product.alphabetized)` since this would combine the orderings.
      .includes(:product, order: :user)
      .joins(instrument: :schedule)
      .merge(Schedule.positioned)

```